pkg/gitops/deploykey/deploykey.go

```diff
@@ -0,0 +1,38 @@
+package deploykey
+
+import (
+	"context"
+	"os"
+
+	"github.com/kris-nova/logger"
+	"github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+)
+
+type GitProvider interface {
+	Put(ctx context.Context, fluxSSHKey flux.PublicKey) error
+	Delete(ctx context.Context) error
+}
+
+func ForCluster(cluster *v1alpha5.ClusterConfig) GitProvider {
+	if githubToken := os.Getenv(EnvVarGitHubToken); githubToken != "" {
+		km := &GitHubProvider{
+			cluster:     cluster.Metadata,
+			githubToken: githubToken,
+		}
+
+		if git := cluster.Git; git != nil {
+			if repo := git.Repo; repo != nil {
+				km.repoURL = repo.URL
+			}
+
+			km.readOnly = git.ReadOnly
+		}
+
+		return km
+	}
+
+	logger.Info("GITHUB_TOKEN is not set. Please set it so that eksctl is able to create and delete GitHub deploy key from Flux SSH public key")
```

Comment:
We should log this only if the URL contains `github.com`.

Reply:
Sounds good! Addressed in 169f2a1e

Reply:
Sounds good! Addressed in 169f2a1e

Reply:
4b1ceca7 should be better and correct.